### ISSUE TITLE
[Reproduction Issue] Experimentation results mismatch with the results showing on the paper

### ISSUE THREAD

**Issue by zjc664656505** [waynewu6250/LABAN#3]
Dear Authors,

We are working on the multi-intent detection project now and running your code as our baseline result.

However, during our experimentation, we found using your experimentation setting, we could never reach the accuracy and F1 results you mentioned in your paper. **For example, in the MixATIS dataset, you show your result like this**:

Laban MixATIS Normal:
F1         ACC
0.958     0.889

Laban MixATIS ZSL:
F1-a    F1-s     F1-u
0.544   0.471    0.451

**By experimenting with your work, the result shows on our end in MixATIS dataset**:
1. MixATIS:
   1. Normal Mode:
      1. Testing Acc: P = 0.8910, R = 0.8944, F1 = 0.8927, Accuracy:  0.7294685990338164
   2. Zero Shot Mode:
      1. Label_num |  Ratio | Testing: P R F1 Acc          | Seen: P R F1 Acc             | Unseen: P R F1 Acc
         1. 16        15    | 0.6634 0.5275 0.5739 0.02753 | 0.6597 0.8948 0.7418 0.37922 | 0.0594 0.0594 0.0594 0.05922
         2. 14        12    | 0.7209 0.6624 0.6736 0.24409 | 0.7993 0.8575 0.8192 0.82621 | 0.3341 0.3402 0.3294 0.26538
         3. 13        10    | 0.5462 0.7239 0.6027 0.02586 | 0.6454 0.8946 0.7271 0.37735 | 0.3539 0.4770 0.3872 0.16085
         4. 12        8     | 0.4673 0.6122 0.5146 0.04372 | 0.6303 0.8266 0.6922 0.42427 | 0.2508 0.3335 0.2678 0.09075
   3. Few Shot Mode:
      1. Few_shot_ratio  Testing Acc
         1. 0.1             P = 0.9052, R = 0.8880, F1 = 0.8966, Accuracy:  0.7185990338164251

If it is possible, could you explain why there is such kind of large result mismatch gap between the experimentation result on our end and the result on your end?

Is there anything different between the model you published on the GitHub with the model you used on your local machine?

Please let us know. Thanks.

**Comment by waynewu6250** [comment]
We report the overall average score based on many intent splits of different random seeds. There could be some variations between each split (Some may be large due to seen and unseen intent correlations). Here is only the sample code for just one split for zero-shot setting and we can get the similar results for FSPS if you run python bert_zsl.py train/test directly.
But scores in normal mode you run are a little bit strange. Please double check if you have trained the model till convergence.

**Comment by zjc664656505** [comment]
Hello,

I have trained the normal model until convergence using both 50 epochs and 100 epochs. However, neither of these options will generate the same results shown on the paper. It seems a little bit weird. We double-checked the model and the data we used for many times, and we have three researchers experimented with your model in the normal mode using the mixatis data, but we all receive a similar accuracy score of around .72. Do you know why this is happening?

Thanks.

**Comment by waynewu6250** [comment]
I'm not sure which part you miss. We have put a sample log which we run normal mixatis on our system: https://github.com/waynewu6250/LABAN/blob/main/laban_normal_atis.txt
This single run has F1 = 0.9552, Accuracy=0.8931
For zero-shot, you may also check some error files generated by our zero-shot model in error_analysis/ for debugging.

**Comment by Rain9876** [comment]
Hello,

I am also confused about the results as well. There are 3 points I want to discuss

1. It's true if I run the code directly, and I get the result as your said. However, the results of using processed MixATIS data through train_data.py file ("python3 train_data.py -d mixatis") looks like this

====== epoch 15 / 15: ======
100%|█████████████████████████████████████████| 412/412 [01:03<00:00,  6.44it/s]
Average train loss: 0.0327 
P = 0.9987, R = 0.9980, F1 = 0.9984
Accuracy:  0.9933900623005623
Average val loss: 5.8393 
P = 0.9213, R = 0.8949, F1 = 0.9079
Accuracy:  0.7765700483091788

Best total val loss: 4834.9385
Best Test Accuracy: 0.7814

It seems the data used for experiment is not the same or is valid set from MixATIS dataset.

2.  In the code, it seems that validation set is not used in MixATIS and MixSNIPS. However, it is more reasonable to choose the best model by best validation accuracy rather than test accuracy through all epochs. Because test set should never be touched.


3. ATIS json file is not provided, and using data format of SNIPS and raw ATIS data from AGIF. The results of ATIS is also not as good as paper said.


Thanks
